src/Silverback.Integration/Messaging/Diagnostics/ActivityConsumerBehavior.cs

```diff
@@ -33,34 +31,17 @@ public ActivityConsumerBehavior(IInboundLogger<ActivityConsumerBehavior> logger)
         public int SortIndex => BrokerBehaviorsSortIndexes.Consumer.Activity;
 
         /// <inheritdoc cref="IConsumerBehavior.HandleAsync" />
-        [SuppressMessage("", "CA1031", Justification = Justifications.ExceptionLogged)]
         public async Task HandleAsync(
             ConsumerPipelineContext context,
             ConsumerBehaviorHandler next)
         {
             Check.NotNull(context, nameof(context));
             Check.NotNull(next, nameof(next));
 
-            using var activity = new Activity(DiagnosticsConstants.ActivityNameMessageConsuming);
-
-            try
-            {
-                activity.InitFromMessageHeaders(context.Envelope.Headers);
-            }
-            catch (Exception ex)
```

Comment:
I don't think it is a good pattern to catch theses, since I don't see any obvious way this could fail.

Reply:
Isn't it possible to cause it to fail submitting a wrong value in the related header?

I don't think it should prevent the message to be consumed in that case.

Reply:
Also, I believe that the `catch` was added later on because it happened that it failed to initialize the `Activity`...but it was a very long time ago and I'm not sure.

Reply:
I just could no see any "normal" way this could fail, even with non compliant values in the header. But I agree, that the impossiple cases always happen at some point. So should I add the try...catch around all the activitiy initialization?

Reply:
If you know it doesn't fail you can leave it alone. I trust you. 😉